gtda/diagrams/features.py

```diff
@@ -590,10 +596,13 @@ class PersistentImage(BaseEstimator, TransformerMixin):
     _hyperparameters = {'sigma': [numbers.Number, (1e-16, np.inf)],
```

Comment:
The validity of the weight_function should be checked here as well.

Reply:
True. I don't yet know how. There are some properties I would like it to respect:
- zero on the diagonal,
- positive.
I will ask Umberto,
but do you have an idea?  Sampling a few points?